Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -132,37 +130,55 @@ function Initialize-AzureSubscription {
         } else {
             # Else, this is AzureRM.            
             try {
-                if (Get-Command -Name "Add-AzureRmAccount" -ErrorAction "SilentlyContinue") {                    
+                if (Get-Command -Name "Add-AzureRmAccount" -ErrorAction "SilentlyContinue") {
```

Comment:
Can you do Disable-AzureRmContextAutosave in initialize to ensure the context remains only at process.